Endzone.Umbraco.Extensions/PublishedContentExtensions/Media.cs

```diff
@@ -21,21 +22,40 @@ public static IEnumerable<IPublishedContent> GetMultipleTypedMedia(this IPublish
             if (!content.HasValue(property))
                 return Enumerable.Empty<IPublishedContent>();
 
-            var imageIds = content.GetPropertyValue<string>(property).Split(new[] { "," }, StringSplitOptions.RemoveEmptyEntries);
-            var umbracoHelper = new UmbracoHelper(UmbracoContext.Current);
+            var contentValue = content.GetPropertyValue(property);
 
-            return umbracoHelper.TypedMedia(imageIds);
+            if (contentValue is IEnumerable<IPublishedContent>)
+            {
+                return (IEnumerable<IPublishedContent>)contentValue;
+            }
+            else if (contentValue is string)
+            {
+                // use old comma separated list
+                var imageIds = contentValue.ToString().Split(new[] { "," }, StringSplitOptions.RemoveEmptyEntries);
+                var umbracoHelper = new UmbracoHelper(UmbracoContext.Current);
+
+                return umbracoHelper.TypedMedia(imageIds);
+            }
+
+            return Enumerable.Empty<IPublishedContent>();
```

Comment:
In what situation would this get returned?

Reply:
If it's none of the above, which could be the case if there is no value stored for the property (would return null instead).

Reply:
Wouldn't `!content.HasValue` cover that?  Or am I misunderstanding?

Reply:
Good point. I could make the last `else if` an `else` and assume it's always a string.

Reply:
No, its fine - I was just checking my understanding!  I like the check for a string as it makes that explicit.

Just to be pedantic : maybe add a comment to explain the last return (as I'll probably forget this next time I look at it!)